pupil_src/shared_modules/video_capture/base_backend.py

```diff
@@ -63,22 +71,150 @@ class Base_Source(Plugin):
     icon_chr = chr(0xE412)
     icon_font = "pupil_icons"
 
-    def __init__(self, g_pool):
+    @property
+    def pretty_class_name(self):
+        return "Video Source"
+
+    def __init__(
+        self, g_pool, *, source_mode: T.Optional[SourceMode] = None, **kwargs,
+    ):
         super().__init__(g_pool)
         self.g_pool.capture = self
         self._recent_frame = None
         self._intrinsics = None
 
+        # Three relevant cases for initializing source_mode:
+        #   - Plugin started at runtime: use existing source mode in g_pool
+        #   - Fresh start without settings: initialize to auto
+        #   - Start with settings: will be passed as parameter, use those
+        if not hasattr(self.g_pool, "source_mode"):
+            self.g_pool.source_mode = source_mode or SourceMode.AUTO
+
+        if not hasattr(self.g_pool, "source_managers"):
+            # If for some reason no manager is loaded, we initialize this ourselves.
+            self.g_pool.source_managers = []
+
     def add_menu(self):
         super().add_menu()
         self.menu_icon.order = 0.2
 
+    def init_ui(self):
+        self.add_menu()
+        self.menu.label = "Video Source"
+        self.update_menu()
+
+    def deinit_ui(self):
+        self.remove_menu()
+
+    def source_list(self):
+        source_type = "Camera" if self.manual_mode else "Device"
```

Comment:
What's the difference between `"Camera"` and `"Device"`? :thinking:
Also, is it just for UI text?

Reply:
A device offers a collections of cameras.
In manual mode you select cameras, in auto mode you select devices.

That line is just for UI, correct.

Reply:
thanks for the clarification!